src/main/java/seedu/contax/storage/CsvManager.java

```diff
@@ -0,0 +1,93 @@
+    /**
+     * Creates a CsvManager to import to specified {@code model}
+     */
+    public CsvManager(Model model) {
+        this.model = model;
+    }
+
+    /**
+     * Imports a given {@code IndexedCSvFile} to a {@code model}
+     * @param csvFile File to be imported
+     * @return CommandResult upon success
+     * @throws IOException if reading from file fails
+     */
+    public CommandResult importCsv(IndexedCsvFile csvFile) throws IOException {
+        BufferedReader importedCsv = new BufferedReader(new FileReader(csvFile.getFilePath()));
+        String line = "";
+        ArrayList<Integer> skippedLines = new ArrayList<>();
+        int lineCounter = 0;
+        //skip first line as it contains headers
+        importedCsv.readLine();
+
+        while ((line = importedCsv.readLine()) != null) {
+            lineCounter++;
+            String[] importedPerson = line.split(",");
+            try {
+                Person toAddPerson = personParser(importedPerson, csvFile);
+                model.addPerson(toAddPerson);
+                Set<Tag> tags = toAddPerson.getTags();
+                for (Tag tag : tags) {
+                    if (!model.hasTag(tag)) {
+                        model.addTag(tag);
+                    }
+                }
+            } catch (ParseException | DuplicatePersonException e) {
+                skippedLines.add(lineCounter);
+                continue;
+            }
+        }
+        return outputStringBuilder(skippedLines);
+    }
+
+    private Person personParser(String[] importedPerson, IndexedCsvFile csvFile) throws ParseException {
+        Name toAddName = ParserUtil.parseName(importedPerson[csvFile.getNamePositionIndex()]);
+        Phone toAddPhone = ParserUtil.parsePhone(importedPerson[csvFile.getPhonePositionIndex()]);
+        Email toAddEmail = ParserUtil.parseEmail(importedPerson[csvFile.getEmailPositionIndex()]);
+        Address toAddAddress = ParserUtil.parseAddress(importedPerson[csvFile.getAddressPositionIndex()]);
+        String[] tags = importedPerson[csvFile.getTagPositionIndex()].split(";");
+        Set<Tag> toAddTag = ParserUtil.parseTags(Arrays.asList(tags));
+
+        return new Person(toAddName, toAddPhone, toAddEmail, toAddAddress, toAddTag);
+    }
+    private CommandResult outputStringBuilder(ArrayList<Integer> skippedLines) {
+        if (skippedLines.size() > 0) {
+            String skippedLinesString = "";
+            for (int i = 0; i < skippedLines.size(); i++) {
+                skippedLinesString += skippedLines.get(i);
+                if (i != skippedLines.size() - 1) {
+                    skippedLinesString += ", ";
+                }
+            }
+            return new CommandResult(String.format("%s\n%s", ImportCsvCommand.MESSAGE_SUCCESS,
+                    String.format(ImportCsvCommand.MESSAGE_SKIPPED_LINES, skippedLinesString)));
+        } else {
+            return new CommandResult(ImportCsvCommand.MESSAGE_SUCCESS);
+        }
+    }
```

Comment:
Then, if you separate it using a `BiFunction` these can go into your `ImportCsvCommand` class, and parsing will be strictly isolated to the parser package.

Reply:
thanks for the help! resolved with next commit, let me know if got any more areas for improvement